static/app/views/detectors/components/detectorListTable/index.tsx

```diff
@@ -24,63 +29,98 @@ function LoadingSkeletons() {
   ));
 }
 
-function DetectorListTable({detectors, isPending}: DetectorListTableProps) {
+function HeaderCell({
+  children,
+  name,
+  divider,
+  sortKey,
+  sort,
+}: {
+  children: React.ReactNode;
+  name: string;
+  sort: Sort | undefined;
+  divider?: boolean;
+  sortKey?: string;
+}) {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const isSortedByField = sort?.field === sortKey;
+  const handleSort = () => {
+    if (!sortKey) {
+      return;
+    }
+    const newSort =
+      sort && isSortedByField ? `${sort.kind === 'asc' ? '-' : ''}${sortKey}` : sortKey;
+    navigate({
+      pathname: location.pathname,
+      query: {...location.query, sort: newSort},
+    });
+  };
+
   return (
-    <Panel>
+    <ColumnHeaderCell
+      className={name}
+      isSorted={isSortedByField}
+      onClick={handleSort}
+      role="columnheader"
+      as={sortKey ? 'button' : 'div'}
+    >
+      {divider && <HeaderDivider />}
+      {sortKey && <InteractionStateLayer />}
+      <Heading>{children}</Heading>
+      {sortKey && (
+        <SortIndicator
+          aria-hidden
+          size="xs"
+          direction={sort?.kind === 'asc' ? 'up' : 'down'}
+          isSorted={isSortedByField}
+        />
+      )}
+    </ColumnHeaderCell>
+  );
+}
+
+function DetectorListTable({detectors, isPending, sort}: DetectorListTableProps) {
+  return (
+    <PanelGrid>
       <StyledPanelHeader>
-        <Flex className="type">
-          <Heading>{t('Type')}</Heading>
-        </Flex>
-        <Flex className="type">
-          <HeaderDivider />
-          <Heading>{t('Type')}</Heading>
-        </Flex>
-        <Flex className="issue">
-          <HeaderDivider />
-          <Heading>{t('Last Issue')}</Heading>
-        </Flex>
-        <Flex className="assignee">
-          <HeaderDivider />
-          <Heading>{t('Assignee')}</Heading>
-        </Flex>
-        <Flex className="connected-automations">
-          <HeaderDivider />
-          <Heading>{t('Automations')}</Heading>
-        </Flex>
+        <HeaderCell name="name" sortKey="name" sort={sort}>
+          {t('Name')}
+        </HeaderCell>
+        <HeaderCell name="type" divider sortKey="type" sort={sort}>
```

Comment:
Are these `sortKeys` temporary until the API implements more sorting? In the designs I only see the option to sort by Last Issue

Reply:
I added all the ones that the API currently supports. When we get the open periods stuff I'll add in the sorting by last issue, but I think it's good to be able to sort by all the columns we can